src/main/java/nl/tudelft/goalkeeper/parser/queries/ExpressionParser.java

```diff
@@ -0,0 +1,42 @@
+package nl.tudelft.goalkeeper.parser.queries;
+
+import nl.tudelft.goalkeeper.exceptions.UnknownKRLanguageException;
+import nl.tudelft.goalkeeper.parser.results.parts.Expression;
+import swiprolog.language.PrologExpression;
+
+/**
+ * Class that parses expressions.
+ */
+public final class ExpressionParser {
+
+    /**
+     * Prevents instantiation.
+     */
+    private ExpressionParser() { }
+
+    /**
+     * Parses a GOAL query to a GOALkeeper expression.
+     * @param expression Expression to parse.
+     * @return Parsed expression.
+     * @throws UnknownKRLanguageException Thrown when we can't handle the query.
+     */
+    public static Expression parse(krTools.language.Expression expression)
+            throws UnknownKRLanguageException {
+        return getParser(expression).parse(expression);
+    }
+
+    /**
+     * Gets the an instance of the correct parser type.
+     * @param expression Expression to get the type for.
+     * @return Instance of the correct parser.
+     * @throws UnknownKRLanguageException Thrown when we can't handle the query.
+     */
+    static ExpressionParserInterface getParser(krTools.language.Expression expression)
```

Comment:
This would make testing really awful and not much harm could be done by not having this private (since the method does not change the state of anything)